bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -268,6 +279,15 @@ protected void addProvider(Provider<E> provider) {
         }
     }
 
+    /**
+     * Returns the key for a given element
+     *
+     * @param element
+     *            element
+     * @return key (must not be null)
+     */
+    protected abstract K getKey(E element);
```

Comment:
> the name UniqueIdProvider is in conflict with Provider interface, could we change it to UniquelyIdentifiable or something like this?

Sure, I am happy to change the naming.
I will wait for that change as soon as I get a go. The change would  be then part of the PR if we agree.